src/payments-methods/express-helper/payment-methods.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+import React from 'react';
+
+const { registerCheckoutBlock } = wc.blocksCheckout;
+
+/**
+ * Internal dependencies
+ */
+import { getCheckOutFieldsLabel } from '../../utils';
+import { activateChange } from '../../renderAmazonButton';
+import { settings } from '../express/settings';
+import { changeShippingAddressOptions, logOutBannerOptions } from './checkout-blocks';
+
+// Register our checkout Blocks.
+registerCheckoutBlock( changeShippingAddressOptions );
+registerCheckoutBlock( logOutBannerOptions );
+
+/**
+ * The change Payment method component.
+ *
+ * @returns React component
+ */
+const ChangePayment = () => {
+	useEffect( () => {
+		activateChange( 'amazon_change_payment_method', 'changePayment' );
+	}, [] );
+
+	return (
+		<a href="#" className="wc-apa-widget-change" id="amazon_change_payment_method">
+			{ settings.hasPaymentPreferences ? __( 'Change', 'woocommerce-gateway-amazon-payments-advanced' ) : __( 'Select', 'woocommerce-gateway-amazon-payments-advanced' ) }
+		</a>
+	);
+};
+
+/**
+ * Returns a react component and also sets an observer for the onCheckoutValidationBeforeProcessing event.
+ *
+ * @param {object} props
+ * @returns React component
+ */
+const AmazonPayInfo = ( props ) => {
+	const { shippingAddress, setShippingAddress } = props.shippingData;
+
+	const { billingData } = props.billing;
+
+	const { amazonBilling, amazonShipping } = settings.amazonAddress;
+
+	useEffect( () => {
+		const unsubscribe = props.eventRegistration.onCheckoutValidationBeforeProcessing(
```

Comment:
Minor, but i'm not sure if `unsubscribe` is a clear name for this, perhaps i'm missing something.